src/github_query_issues.py

```diff
@@ -165,33 +190,37 @@ def process_issues(issues: List[dict], org_name: str, repo_name: str) -> List[di
         md_filename_base = f"{issue['number']}_{issue['title'].lower()}.md"
         sanitized_md_filename = sanitize_filename(md_filename_base)
 
-        issue_data = {
-            "Number": issue['number'],
-            "Owner": org_name,
-            "RepositoryName": repo_name,
-            "Title": issue['title'],
-            "State": issue['state'],
-            "URL": issue['html_url'],
-            "Body": issue['body'],
-            "CreatedAt": issue['created_at'],
-            "UpdatedAt": issue['updated_at'],
-            "ClosedAt": issue['closed_at'],
-            "MilestoneNumber": milestone_number,
-            "MilestoneTitle": milestone_title,
-            "MilestoneHtmlUrl": milestone_html_url,
-            "Labels": label_names,
-            "PageFilename": sanitized_md_filename
-        }
-        issue_list.append(issue_data)
-
-    return issue_list
-
-
-if __name__ == "__main__":
-    print("Downloading issues from GitHub started")
-
-    # Get environment variables set by the controller script
-    user_token = os.getenv('GITHUB_TOKEN')
+        # Define Issue object as it follows the Issue dataclass
+        issue_object = Issue(
+            number=issue['number'],
+            owner=org_name,
+            repository_name=repo_name,
+            title=issue['title'],
+            state=issue['state'],
+            url=issue['html_url'],
+            body=issue['body'],
+            created_at=issue['created_at'],
+            updated_at=issue['updated_at'],
+            closed_at=issue['closed_at'],
+            milestone_number=milestone_number,
+            milestone_title=milestone_title,
+            milestone_html_url=milestone_html_url,
+            labels=label_names,
+            page_filename=sanitized_md_filename
+        )
+
+        # Convert Issue object to dictionary, because JSON does not support dataclasses
+        issue_dict = issue_object.to_dict()
```

Comment:
Here, why create an Issue and immediately convert it back to dirt?
Why not create a list of Issue instances and return them?

Reply:
Error: `Object of type Issue is not JSON serializable`. Keeping it for now, up to debate, how to deal with it.